pkg/api/latest/latest.go

```diff
@@ -18,125 +18,101 @@ package latest
 
 import (
 	"fmt"
-	"sort"
-	"strings"
 
+	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/meta"
 	"k8s.io/kubernetes/pkg/api/registered"
 	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/api/v1"
 	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/sets"
 )
 
-var (
-	allGroups = GroupMetaMap{}
-	// Group is a shortcut to allGroups.Group.
-	Group = allGroups.Group
-	// RegisterGroup is a shortcut to allGroups.RegisterGroup.
-	RegisterGroup = allGroups.RegisterGroup
-	// GroupOrDie is a shortcut to allGroups.GroupOrDie.
-	GroupOrDie = allGroups.GroupOrDie
-	// AllPreferredGroupVersions returns the preferred versions of all
-	// registered groups in the form of "group1/version1,group2/version2,..."
-	AllPreferredGroupVersions = allGroups.AllPreferredGroupVersions
-	// IsRegistered is a shortcut to allGroups.IsRegistered.
-	IsRegistered = allGroups.IsRegistered
-)
+// ExternalVersions lists all known external versions for this group from most preferred to least preferred
+var ExternalVersions = []unversioned.GroupVersion{v1.SchemeGroupVersion}
```

Comment:
> Looks like a misunderstanding of kind versus resource, but this simply maintains existing behavior

I'm not following... what does kind or resource have to do with groupversions? Is the intent of this to list all available versions, and in `init()` to limit it to what is actually registered?